### PR TITLE
Oran gyro function

### PR DESCRIPTION
**Pull request by Pastyguy**
Compiles but is untested
 need review of math
needs to be compiled on other pc besides mine to confirm that the sensor libraries work as intended

[X] Compiles on desktop
[X] Complies on Teensy
#### Additions

Created a magnetometer class that allows the reading of the magnetometer
usefull functions
     getDirection();
or
     getDirectionRaw();

```
 getHeading():
```

### CODE REVIEW

**Comment by blhough** [comment]
please exclude the changes to PIDControl and robotIO that do not pertain to your magnetometer.
Also the VS project file is still corrupted as you can see in the file changes tab of the pull request

File: micromouse/micromouse.vcxproj [hunk 1]
```diff
@@ -0,0 +1,79 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (blhough) [inline micromouse/micromouse.vcxproj:1]: This FIle should not exists, should be at root

File: micromouse/RobotIO.cpp [hunk 2]
```diff
@@ -183,32 +187,154 @@ namespace Micromouse
...
 		leftMotor.brake();
 		rightMotor.brake();
-#endif
 	}
 
 	// ## NOT YET IMPLEMENTED ##
-	void RobotIO::moveForward()
+	void RobotIO::moveForward(float node)
 	{
+		int countsNeeded=node*4807;
```
Comment (blhough) [inline micromouse/RobotIO.cpp:203]: many of these changes in this file are unrelated to the magnetometer class

**Comment by Pastyguy** [comment]
So uncommitted all files that are not adafriuit,magnatometer.h .cpp

File: micromouse/Magnetometer.cpp [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+	}
+
+
+
+
+
+
+	void Magnetometer::initSensor()
+	{
+		initDirection();
```
Comment (blhough) [inline micromouse/Magnetometer.cpp:62]: cannot init the direction before the sensor is setup

File: micromouse/Magnetometer.cpp [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+	#endif
+
+	}
+
+
+
+
+
+
+	Magnetometer::Magnetometer(const int MAGNETOMETER_NINE_DOF_SDA_PIN, const int MAGNETOMETER_NINE_DOF_SCL_PIN)
```
Comment (blhough) [inline micromouse/Magnetometer.cpp]: you pass in the parameters but do nothing with them

File: MicroMouse-2016.vcxproj.filters [hunk 5]
```diff
@@ -1,20 +1,36 @@
...
+    <ClCompile Include="micromouse\Node.cpp" />
+    <ClCompile Include="micromouse\Path.cpp" />
+    <ClCompile Include="micromouse\RobotIO.cpp" />
+    <ClCompile Include="micromouse\IRSensor.cpp" />
+    <ClCompile Include="micromouse\Vector.cpp" />
+    <ClCompile Include="micromouse\VirtualMaze.cpp" />
+    <ClCompile Include="micromouse\Magnetometer.cpp" />
   </ItemGroup>
   <ItemGroup>
+<<<<<<< HEAD
```
Comment (blhough) [inline MicroMouse-2016.vcxproj.filters]: corrupted
